core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -46,6 +47,35 @@ public static MetricsConfig getDefault() {
     return spec;
   }
 
+  public static Map<String, String> updateProperties(Map<String, String> props, List<String> deletes,
```

Comment:
Seems like we don't need it as public?

I'd also consider a utility class `MetricsConfigUtil` to keep `MetricsConfig` simple. We will need a few helper methods for metrics configs and it may be a good time to introduce a utility class now. That way, we won't have to reason about compatibility when we move to column ids.

Reply:
I think this needs to be public since it's called from another class no?

Reply:
This class as well as `SchemaUpdate` (where the new method is used) are in `org.apache.iceberg` package. I don't think users are supposed to interact with this logic, so I'd make the method package-private like we do in other places.